api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,6 +87,11 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// ClientCertRef defines the reference to a Kubernetes Secret that contains
+	// the client certificate and private key for Envoy to use when connecting to
+	// backend services and external services, such as ExtAuth, ALS, OpenTelemetry, etc.
+	ClientCertRef *gwapiv1.SecretObjectReference `json:"clientCertRef,omitempty"`
```

Comment:
Do we want a container here for more TLS-related configurations like these mentioned in https://github.com/envoyproxy/gateway/issues/2901 ?

Reply:
+1 for container with additional settings

Reply:
This is not the ideal location for backend services' TLS parameters since they may be different per backend. But we may still need a container for other client-side TLS options.

Reply:
In the long term, I expect that this will be part of BTLS Policy, with all the flexibility required per-backend. In the meanwhile, this could be reasonable mitigation...